src/otx/core/model/module/detection.py

```diff
@@ -133,3 +133,11 @@ def test_step(self, inputs: DetBatchDataEntity, batch_idx: int) -> None:
     def lr_scheduler_monitor_key(self) -> str:
         """Metric name that the learning rate scheduler monitor."""
         return "train/loss"
+
+    def _get_state_dict_from_ckpt(self, ckpt: dict) -> dict:
```

Comment:
Do we need to change this as well? Would you please add unit tests in this PR?

Reply:
is this method only needed for detection and segmentation tasks? If it is needed by all the tasks, I see a duplicated pattern in the methods that might potentially be refactored, something like:

```python
    def _get_state_dict_from_ckpt(self, ckpt: dict, condition: bool) -> dict:
        """Get the state_dict, supporting the backward compatibility."""
        state_dict = super()._get_state_dict_from_ckpt(ckpt)
        for key in list(state_dict.keys()):
            if condition:
                state_dict.pop(key)
        return state_dict
```

and in task-based implementation, something like

```python
self._get_state_dict_from_ckpt(ckpt=ckpt, condition=key.startswith("model.model.ema_"))
```

This is a bit minor comment though. You could skip if you disagree

Reply:
> Do we need to change this as well? Would you please add unit tests in this PR?

For the detection and instance segmentation tasks, there were additional chekcpoint keys for the EMA model. So, we need to eliminate theses keys to successfully load the checkpoint. I'll add the unit-test, thanks.

Reply:
Rename the function: https://github.com/openvinotoolkit/training_extensions/pull/2812/commits/1f55383109a5342291c7c8b1d24d9bb3893f66c1